### PR TITLE
Add comma to population and add country code in the api call

### PR DESCRIPTION
**Pull request by sangeethailango**

### CODE REVIEW

File: lib/country_finder.ex [hunk 1]
```diff
@@ -31,15 +31,26 @@ defmodule CountryFinder do
       flag_png = get_in(country_map, ["flags", "png"])
       region = country_map["region"]
       capital = country_map["capital"]
-      population = country_map["population"]
+      population = add_comma(country_map["population"])
+      country_code = get_in(country_map, ["cca3"])
```
Comment (nittin-shankar) [inline lib/country_finder.ex]:
```suggestion
      country_code = country_map["cca3"]
```
Don't use `get_in` function when it is not listed

File: lib/country_finder.ex [hunk 2]
```diff
@@ -31,15 +31,26 @@ defmodule CountryFinder do
...
         region: region,
         capital: capital,
-        population: population
+        population: population,
+        country_code: country_code
       }
     end)
   end
+
+  def add_comma(integer) do
```
Comment (nittin-shankar) [inline lib/country_finder.ex]: Good job writing the function here 💯

File: lib/country_finder.ex [hunk 3]
```diff
@@ -31,15 +31,26 @@ defmodule CountryFinder do
       flag_png = get_in(country_map, ["flags", "png"])
       region = country_map["region"]
       capital = country_map["capital"]
-      population = country_map["population"]
+      population = add_comma(country_map["population"])
```
Comment (nittin-shankar) [inline lib/country_finder.ex]: In the struct, store it just as an integer. Use the function in template. Also, it'll be better if rename the function from `add_comma` to `format_population`. Write some documentation for the function too.

**Review by nittin-shankar — CHANGES_REQUESTED** [review]
Good job completing the task. Please take a look at review comments.

File: lib/country_finder.ex (hunk 1 above)
Comment (sangeethailango) [reply lib/country_finder.ex]: Done

File: lib/country_finder.ex (hunk 3 above)
Comment (sangeethailango) [reply lib/country_finder.ex]: Can i write documentation above the function, using @doc?

Comment (nittin-shankar) [reply lib/country_finder.ex]: Yes, please do that!!

**Review by nittin-shankar — CHANGES_REQUESTED** [review]
@sangeethailango before you push other changes. Please format the code and push it.

File: lib/country_finder.ex [hunk 4]
```diff
@@ -45,6 +45,15 @@ defmodule CountryFinder do
     end)
   end
 
+  @doc """
+     This function is to add comma every 3 digits of a integer from the right.
+
+     Integer.to_charlist() -> Converting integer to charlist as Enum.reverse expects list of intger.
```
Comment (nittin-shankar) [inline lib/country_finder.ex]: We don't need this much additional information on the documentation. You're formatting it with a separator called **thousands seperator**.  You don't need to explain how the function works in documentation.

Just something like below is enough:
_Formats the given integer with thousands separator._

And you can also provide with a few examples as to how this function works. Have you seen there are examples of almost all functions in the `Enum` module. Examples allow for better grasping.

**Review by nittin-shankar — CHANGES_REQUESTED** [review]
Good attempt in writing the documentation. But it could be even better!!!

**Review by nittin-shankar — APPROVED** [review]
Good job but I'll be closing this PR as I merged PR of @kodi6

**Pull request closed without merging**